zarnitsa/DataAugmenterNLP.py

```diff
@@ -110,70 +112,79 @@ def _check_synset(self, name):
-            wget.download(
-                "http://nlpgrid.seas.upenn.edu/PPDB/eng/ppdb-2.0-tldr.gz"
-            )
+            wget.download("http://nlpgrid.seas.upenn.edu/PPDB/eng/ppdb-2.0-tldr.gz")
         else:
-            raise KeyError(
-                f"Synset {name} is unknown! Load manually or fix the name"
-            )
+            raise KeyError(f"Synset {name} is unknown! Load manually or fix the name")
 
-    def augment_column_wordnet(self, text: str) -> str:
+    def augment_wordnet(self, text: str) -> str:
         """Augment column data using wordnet synset."""
         if not self.__aug_wdnt:
             print("Load WordNet synset")
             self.__aug_wdnt = naw.SynonymAug(aug_src="wordnet")
         text = self.__aug_wdnt.augment(text)
         return text
 
-    def augment_column_ppdb(self, text: str) -> str:
+    def augment_ppdb(self, text: str) -> str:
         """Augment column data using ppdb synset."""
         if not self.__aug_ppdb:
             print("Load PPDB synset")
             self._check_synset("ppdb")
             self.__aug_ppdb = naw.SynonymAug(
                 aug_src="ppdb",
-                model_path=self.__class_local_path
-                + "/internal_data/ppdb-2.0-tldr",
+                model_path=self.__class_local_path + "/internal_data/ppdb-2.0-tldr",
             )
         text = self.__aug_ppdb.augment(text)
         return text
 
     @staticmethod
-    def augment_column_word_emb(
-        text: str, words_and_vectors: dict, reps=1
+    def augment_word_emb(
+        text: str,
+        vocab_name=None,
+        nlp=None,
+        reps=1,
+        topn=5,
     ) -> str:
         """Augment column data using embeddings."""
-        if not isinstance(words_and_vectors, dict):
-            raise TypeError("Var words_and_vectors must be dict!")
-        elif not words_and_vectors:
-            raise ValueError("Var words_and_vectors is empty!")
+        if not nlp:
+            if not vocab_name:
+                print("Use spacy en_core_web_sm vocab!")
+                nlp = spacy.load("en_core_web_md")
+            else:
+                nlp = spacy.load(vocab_name)
+
+        def get_most_similar(word, topn=topn):
+            """Get topn words for defined word"""
+            word = nlp.vocab[str(word)]
+            queries = [
+                w
+                for w in word.vocab
+                if w.is_lower == word.is_lower
+                and np.count_nonzero(w.vector)
+            ]
+            by_similarity = sorted(
+                queries, key=lambda w: word.similarity(w), reverse=True
+            )
+            return [
+                w.lower_.capitalize()
+                if not word.lower_.islower() and not word.lower_.isupper()
+                else w.lower_
+                if word.islower()
+                else word.lower_.upper()
+                for w in by_similarity[: topn + 1]
+                if w.lower_ != word.lower_
+            ]
```

Comment:
I don't think that nested ternary operator is a good style, expand it to achieve better readability

Reply:
ok